pkg/csi_driver/controller.go

```diff
@@ -104,28 +104,26 @@ func (s *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolu
 			return nil, status.Error(codes.AlreadyExists, err.Error())
 		}
 	} else {
-
-		// Get the reserved-IPV4-cidr from the request. If value is not provided, reservedIPV4Range is defaulted to ""
+		// If we are creating a new instance, we need to evaluate the reservedIPRange range if reserved-ipv4-cidr is provided
+		// If the param was not provided, we default reservedIPRange to "" and cloud provider takes care of the allocation
 		reservedIPV4CIDR, ok := req.GetParameters()[paramReservedIPV4CIDR]
 		if ok {
-
-			// Prevents lock acquisition in case the CIDR is invalid
 			validCIDR := s.config.ipAllocator.ValidateCIDR(reservedIPV4CIDR)
 			if !validCIDR {
-				return nil, err
+				return nil, fmt.Errorf("The provided reserved-ipv4-cidr %s evaluated to be invalid after parsing", reservedIPV4CIDR)
 			}
 
 			reservedIPRange, err := s.reserveIPV4Range(ctx, newFiler, reservedIPV4CIDR)
+
+			// Possible cases are 1) CreateInstanceAborted, 2)CreateInstance running in background
+			// In either cases, the ListInstances responses would consist of the updated result
```

Comment:
"the ListInstances response will contain the reservedIPRange if the operation was started"